File: clients/fsharp/Model/DebugData.fs
#nowarn "0058"

namespace SpbAiChamp
namespace SpbAiChamp.Model

open SpbAiChamp

/// TODO - Document
type DebugData = struct end with

    /// Write DebugData to writer
    member this.writeTo(writer: System.IO.BinaryWriter) =
        ()

    /// Read DebugData from reader
    static member readFrom(reader: System.IO.BinaryReader) = new DebugData()

File: clients/rust/src/model/debug_data.rs
use super::*;

/// TODO - Document
#[derive(Clone, Debug, PartialEq, Eq, Hash)]
pub struct DebugData {
}

impl trans::Trans for DebugData {
    fn write_to(&self, writer: &mut dyn std::io::Write) -> std::io::Result<()> {
        Ok(())
    }
    fn read_from(reader: &mut dyn std::io::Read) -> std::io::Result<Self> {
        Ok(Self {
        })
    }
}